hs_discover/src/components/Search.vue

```diff
@@ -335,10 +336,22 @@ export default {
     // Allow manual querystring manipulation of freetext search via URL
     if (document.getElementById('qstring').value.trim() !== '') {
       this.searchtext = document.getElementById('qstring').value.trim();
+      this.prefiltered = true;
     }
     // Allow exact subject search via URL particularly for Resource Landing Page
     if (document.getElementById('prefillsubject').value.trim() !== '') {
-      this.subjectFilter = Array(decodeURIComponent(document.getElementById('prefillsubject').value.trim()));
+      const urlEncodedSubj = document.getElementById('prefillsubject').value.trim();
```

Comment:
line 343 can be moved up before the if condition on line 342 so urlEncodedSubj variable can be used in the if condition.